wallet/src/service/Offers.ts

```diff
@@ -41,6 +44,41 @@ type GiveOrWantEntries = {
   };
 };
 
+const CapDataShape = harden({ body: M.string(), slots: M.array() });
+
+const theOne = <T>(array: Array<T> | undefined) => {
```

Comment:
consider putting this in a utils module. eventually we might want it in `@agoric/internal` but it's pretty lightweight to copy around

Reply:
I don't think this abstraction is appropriate for this use-case, explained in my comment below. Is this motivated rather by some other code in the sdk?

Reply:
You're right. I misread the error conditions for https://github.com/Agoric/wallet-app/pull/125#discussion_r1327626461

There are other cases where an array length other than 1 is an error, and the utility is useful for those, but this spot want `undefined` in that case.